src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -503,4 +504,21 @@ public void testTemplateArrayAccess() throws Exception {
 		CtMethod<?> m2 = resultKlass.getMethod("method2");
 		assertEquals("java.lang.System.out.println(\"second\")", m2.getBody().getStatement(0).toString());
 	}
+
+	@Test
+	public void testStatementTemplateRootSubstitution() throws Exception {
+		//contract: the template engine supports substitution of root element
```

Comment:
what do you mean by "root element"?

in `new SubstituteRootTemplate(param).apply(resultKlass);` do you mean `param` or `resultKlass` or something else?